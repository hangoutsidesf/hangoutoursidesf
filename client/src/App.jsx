import React, { Component } from 'react';

import MapView from './mapView';
import Interface from './Interface';
import fetchParklets from '../utils/fetchParklets';
import fakeFilterFlags from '../utils/fakeFilterFlags';

const PARKLETS_ENDPOINT = 'https://data.sfgov.org/resource/6a7x-cttf.json';

class App extends Component {
  constructor() {
    super();
    this.state = {
      activeParklets: [],
      hiddenParklets: [],
      error: '',
      openFilter: false,
      foodFilter: false,
      wifiFilter: false,
    };

    this.handleFilters = this.handleFilters.bind(this);
  }

  componentDidMount() {
    // IMPORTANT: Until db is implemented, fake data will be used to generate flags
    // for a parklet that is open, has coffee, or wifi by RNG, this will change for
    // every refresh of the page
    fetchParklets(PARKLETS_ENDPOINT)
      .then(fakeFilterFlags)
      .then(data => this.setState({ activeParklets: data }))
      .catch(err => this.setState({ error: err }));
  }

  refreshParkletDisplay(parklets) {
    const { openFilter, foodFilter, wifiFilter } = this.state;
    const hasFilter = openFilter || foodFilter || wifiFilter;
    const activeParklets = [];
    const hiddenParklets = [];

    parklets.forEach((parklet) => {
      const { open, food, wifi } = parklet;

      if ((!open && openFilter) || (!food && foodFilter) || (!wifi && wifiFilter)) {
        hiddenParklets.push(parklet);
      } else {
        activeParklets.push(parklet);
      }
    });

    this.setState({ activeParklets, hiddenParklets });
  }

  handleFilters(type) {
    const { activeParklets, hiddenParklets } = this.state;
    const parklets = [...activeParklets, ...hiddenParklets];

    this.setState({ [type]: !this.state[type] }, () => {
      this.refreshParkletDisplay(parklets);
    });
  }

  render() {
    const { error, activeParklets } = this.state;

    return (
      <div id="home">
        <MapView error={error} parklets={activeParklets} />
        <Interface handleFilters={this.handleFilters} />
      </div>
    );
  }
}

export default App;
