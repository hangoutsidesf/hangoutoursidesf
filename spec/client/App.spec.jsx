import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../client/src/App';

describe('App component', () => {
  xtest('App should match snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
