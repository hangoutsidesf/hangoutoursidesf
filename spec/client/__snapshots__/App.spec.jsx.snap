// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App should match snapshot 1`] = `
<div>
  let's hang out in san francisco!
</div>
`;
